skbio/embedding/_protein.py

```diff
@@ -7,10 +7,28 @@
 # ----------------------------------------------------------------------------
 from skbio.sequence import Protein
 from skbio.embedding._embedding import SequenceEmbedding
+from skbio.embedding._embedding import SequenceVector
+from skbio.embedding._embedding import _repr_helper
+from skbio.stats.ordination import OrdinationResults
+from scipy.spatial.distance import pdist, squareform
 from skbio.util import get_data_path
 from pathlib import Path
 import pandas as pd
 import numpy as np
+from typing import List
+
+
+def _validate_protein(sequence):
+    if isinstance(sequence, Protein):
+        sequence = str(sequence)
+    elif isinstance(sequence, str):
+        if " " in sequence:
+            sequence = sequence.replace(" ", "")
+
+        # perform a check to make sure the sequence is a valid
+        # protein sequence
+        _ = Protein(sequence)
+    return sequence
 
 
 class ProteinEmbedding(SequenceEmbedding):
```

Comment:
The parameter `sequence` may be "array_like" instead of just "ndarray", such that plain Python lists can be supported. The code can add `sequence = np.asarray(sequence)` to convert lists into ndarray, or leave as-is if already array.